packages/web/src/session/session.ts

```diff
@@ -73,37 +111,46 @@ export function getCurrentSessionState({ useLocalStorage = false, forceStoreRead
 // (Only exported for testing purposes.)
 export function updateSessionStatus({
 	forceStore,
-	useLocalStorage = false,
 	forceActivity,
+	shadow = undefined,
 }: {
 	forceActivity?: boolean
 	forceStore: boolean
-	useLocalStorage: boolean
-}): void {
-	let sessionState = getCurrentSessionState({ useLocalStorage, forceStoreRead: forceStore })
+	shadow?: boolean
+}): SessionState {
+	let sessionState = getCurrentSessionState({ forceDiskRead: forceStore })
 	let shouldForceWrite = false
 	if (!sessionState) {
 		// Check if another tab has created a new session
-		sessionState = getCurrentSessionState({ useLocalStorage, forceStoreRead: true })
+		sessionState = getCurrentSessionState({ forceDiskRead: true })
 		if (!sessionState) {
 			sessionState = createSessionState()
 			recentActivity = true // force write of new cookie
 			shouldForceWrite = true
 		}
 	}
 
+	if (sessionState.shadow !== shadow) {
+		sessionState.shadow = shadow;
+		shouldForceWrite = true;
+	}
+
 	eventTarget?.emit('session-changed', { sessionId: sessionState.id })
 
 	if (recentActivity || forceActivity) {
 		sessionState.expiresAt = Date.now() + SESSION_INACTIVITY_TIMEOUT_MS
-		if (useLocalStorage) {
-			setSessionStateToLocalStorage(sessionState, { forceStoreWrite: shouldForceWrite || forceStore })
-		} else {
-			renewCookieTimeout(sessionState, cookieDomain, { forceStoreWrite: shouldForceWrite || forceStore })
+
+		// TODO: review if this check makes sense
```

Comment:
I was thinking exactly the same but was unsure if it could be removed. I would go for it, and since there are two of us who think the same, I guess it's safe.